postproengine/wake_meandering.py

```diff
@@ -0,0 +1,398 @@
+            ncfile   = entry['ncfile']
+            group    = entry['group']
+            self.yhub     = entry['yhub']
+            self.zhub     = entry['zhub']
+            diam     = entry['diam']
+            Ct       = entry['Ct']
+            Uinf     = entry['Uinf']
+            method   = entry['method']
+            savefile = entry['savefile']
+            varnames = ['velocityx','velocityy','velocityz']
+            self.output_dir =  entry['output_dir']
+
+            if not isinstance(iplanes, list): iplanes = [iplanes,]
+            #Get all times if not specified 
+            filelist = []
+            for fileiter in range(0,len(ncfile)):
+                filelist.append(ncfile[fileiter])
+
+            udata = {}
+            xc = {}
+            timevec = ppsample.getVar(ppsample.loadDataset(ncfile), 'time')
+            if trange==[]:
+                # load entire file
+                self.db = ppsamplexr.getFullPlaneXR(ncfile,len(timevec),timevec[1]-timevec[0],groupname=group)
+                y = np.asarray(self.db['y'].data)
+                z = np.asarray(self.db['z'].data)
+                t = np.asarray(np.array(self.db['times']).data)
+                for iplane in iplanes:
+                    xc[iplane] = np.asarray(self.db['x'].data[iplane])
+                    udata[iplane] = np.zeros((len(t),len(z),len(y),3))
+                    udata[iplane][:,:,:,0] = np.array(self.db['velocityx'][:,iplane,:,:])
+                    udata[iplane][:,:,:,1] = np.array(self.db['velocityy'][:,iplane,:,:])
+                    udata[iplane][:,:,:,2] = np.array(self.db['velocityz'][:,iplane,:,:])
+            else:
+                find_nearest = lambda a, a0: np.abs(np.array(a) - a0).argmin()
+                iters = [find_nearest(timevec, t) for t in trange]
+                iters = np.arange(iters[0],iters[1]+1)
+                self.db = ppsamplexr.getPlaneXR(ncfile,iters,varnames,groupname=group,verbose=0,includeattr=False,gettimes=True)
+                y = np.asarray(self.db['y'][0][0,:])
+                z = np.asarray(self.db['z'][0][:,0])
+                t = np.asarray(np.array(self.db['times']).data)
+                for iplane in iplanes:
+                    xc[iplane] = np.asarray(self.db['x'][iplane][0,0])
+                    udata[iplane] = np.zeros((len(t),len(z),len(y),3))
+                    for i,tstep in enumerate(iters):
+                        udata[iplane][i,:,:,0] = np.array(self.db['velocityx'][tstep][iplane,:,:])
+                        udata[iplane][i,:,:,1] = np.array(self.db['velocityy'][tstep][iplane,:,:])
+                        udata[iplane][i,:,:,2] = np.array(self.db['velocityz'][tstep][iplane,:,:])
+
+            YY , ZZ = np.meshgrid(y,z)
+            arg=None
+            if method == 'ConstantArea':
+                arg = np.pi * (diam/2.0)**2
+            elif method == 'ConstantFlux':
+                arg = np.pi * (diam/2.0)**2 * Ct * Uinf ** 2
```

Comment:
Add option to specify mean rotor thrust from openfast directly. 

Also, missing a 0.5 factor here

Reply:
Done!